xds/src/test/java/io/grpc/xds/LookasideLbTest.java

```diff
@@ -243,80 +185,183 @@ public int getPriority() {
+    }
+    if (withXdsClientPoolAttributes) {
+      assert !withBootstrap;
+      xdsClientPoolFromResolveAddresses.returnObject(xdsClientFromResolvedAddresses);
+      assertThat(xdsClientPoolFromResolveAddresses.xdsClient).isNull();
+    }
+  }
+
+  // For EDS-only usecase.
+  private void setUpWithBootstrap() throws Exception {
+    withBootstrap = true;
+    Bootstrapper bootstrapper = mock(Bootstrapper.class);
+    AggregatedDiscoveryServiceImplBase serviceImpl = new AggregatedDiscoveryServiceImplBase() {
+      @Override
+      public StreamObserver<DiscoveryRequest> streamAggregatedResources(
+          final StreamObserver<DiscoveryResponse> responseObserver) {
+        responseObservers.offer(responseObserver);
+        @SuppressWarnings("unchecked")
+        StreamObserver<DiscoveryRequest> requestObserver = mock(StreamObserver.class);
+        return requestObserver;
+      }
+    };
+    String serverName = InProcessServerBuilder.generateName();
+    cleanupRule.register(
+        InProcessServerBuilder
+            .forName(serverName)
+            .directExecutor()
+            .addService(serviceImpl)
+            .build()
+            .start());
+    bootstrapChannel = cleanupRule.register(
+        InProcessChannelBuilder
+            .forName(serverName)
+            .directExecutor()
+            .build());
     List<ServerInfo> serverList =
         ImmutableList.of(
             new ServerInfo("trafficdirector.googleapis.com", ImmutableList.<ChannelCreds>of()));
     BootstrapInfo bootstrapInfo = new BootstrapInfo(serverList, Node.getDefaultInstance());
     doReturn(bootstrapInfo).when(bootstrapper).readBootstrap();
 
-    lookasideLb = new LookasideLb(
-        helper, edsUpdateCallback, lbRegistry, localityStoreFactory, loadReportClientFactory,
-        bootstrapper);
-
-    String lbConfigRaw11 = "{}";
-    @SuppressWarnings("unchecked")
-    Map<String, ?> lbConfig11 = (Map<String, ?>) JsonParser.parse(lbConfigRaw11);
-    defaultResolvedAddress = ResolvedAddresses.newBuilder()
-        .setAddresses(ImmutableList.<EquivalentAddressGroup>of())
-        .setAttributes(Attributes.newBuilder().set(ATTR_LOAD_BALANCING_CONFIG, lbConfig11).build())
-        .build();
-  }
+    XdsChannelFactory channelFactory = new XdsChannelFactory() {
+      @Override
+      ManagedChannel createChannel(List<ServerInfo> servers) {
+        assertThat(Iterables.getOnlyElement(servers).getServerUri())
+            .isEqualTo("trafficdirector.googleapis.com");
+        return bootstrapChannel;
+      }
+    };
 
-  @Test
-  public void canHandleEmptyAddressListFromNameResolution() {
-    assertThat(lookasideLb.canHandleEmptyAddressListFromNameResolution()).isTrue();
+    lookasideLb = new LookasideLb(
+        helper, edsUpdateCallback, lbRegistry, localityStoreFactory, bootstrapper, channelFactory);
   }
 
-  @Test
-  public void handleNameResolutionErrorBeforeAndAfterEdsWorkding() throws Exception {
+  // For non EDS-only usecase.
```

Comment:
Bad phrasing `non EDS-only usecase`.

Reply:
non-EDS-only?

Reply:
I think you only need to comment for "EDS only" explanations, as that's something we keep backwardly compatible with. That comment explains the reason why we are doing in that way. For xDS full flow logic, you don't need to mention, as that would be reader's main focus.